src/compile/mod.rs

```diff
@@ -594,6 +519,97 @@ fn create_type_tree(program_tree: &HashMap<Vec<String>, Module>) -> TypeTree {
         .collect()
 }
 
+fn typecheck_programs(
+    type_tree: &TypeTree,
+    output: Vec<Module>,
```

Comment:
The name `output` is non-intuitive here. Is there a more descriptive name we could use?